tpm_quote/src/lib.rs

```diff
@@ -0,0 +1,107 @@
+use error::Error;
+use key::{get_key_from_persistent_handle, TPM_AK_NVINDEX};
+use std::convert::TryInto;
+use tss_esapi::{
+    interface_types::algorithm::HashingAlgorithm,
+    structures::{
+        Attest, AttestInfo, PcrSelectionListBuilder, PcrSlot, Public, QuoteInfo, Signature,
+    },
+    traits::{Marshall, UnMarshall},
+    Context,
+};
+use verify::verify_quote_signature;
+
+pub mod command;
+pub mod convert;
+pub mod error;
+pub mod key;
+pub mod verify;
+
+#[allow(clippy::type_complexity)]
+pub fn get_quote(
+    context: &mut Context,
+    pcr_list: &[u8],
+    nonce: Option<&[u8]>,
+) -> Result<(Vec<u8>, Vec<u8>, Vec<u8>), Error> {
+    let pcr_list = pcr_list
+        .iter()
+        .map(|n| (1 << (*n as u32)).try_into().unwrap())
+        .collect::<Vec<PcrSlot>>();
+
+    let pcr_selection_list = PcrSelectionListBuilder::new()
+        .with_selection(HashingAlgorithm::Sha1, &pcr_list)
+        .with_selection(HashingAlgorithm::Sha256, &pcr_list)
+        .build()?;
+
+    let (ak_handle, ak_pub) = get_key_from_persistent_handle(context, TPM_AK_NVINDEX)?;
+
+    let (attestation_data, signature) =
+        command::attest_pcr(context, &ak_handle, pcr_selection_list, nonce)?;
+
+    Ok((
+        attestation_data.marshall()?,
+        signature.marshall()?,
+        ak_pub.marshall()?,
+    ))
+}
+
+pub fn verify_quote(
+    quote: &[u8],
+    signature: &[u8],
+    public_key: &[u8],
+    nonce: Option<&[u8]>,
+) -> Result<QuoteInfo, Error> {
+    let attestation_data = Attest::unmarshall(quote)?;
+    let signature = Signature::unmarshall(signature)?;
+    let public_key = Public::unmarshall(public_key)?;
+
+    let quote_info = match attestation_data.attested() {
+        AttestInfo::Quote { info } => Ok(info),
+        _ => Err(Error::QuoteError("unexpected nonce in quote".to_owned())),
+    }?;
+    let expected_nonce = verify_quote_signature(&attestation_data, &signature, &public_key)?;
+
+    match nonce {
+        Some(nonce) => {
+            if nonce != expected_nonce {
+                return Err(Error::QuoteError("unexpected nonce in quote".to_owned()));
+            }
+        }
+        None => {
+            if expected_nonce != vec![0xff; 16] {
+                return Err(Error::QuoteError("unexpected nonce in quote".to_owned()));
+            }
+        }
+    }
+
+    Ok(quote_info.to_owned())
+}
+
+#[cfg(test)]
+mod tests {
+    use crate::{get_quote, verify_quote};
+    use tss_esapi::{tcti_ldr::TctiNameConf, Context};
+
+    use test_log::test;
+
+    #[test]
+    #[ignore]
```

Comment:
Could you not ignore the test but make a check in the function. If TPM is enable then return the result of the assert, if not return true. 

Example here: https://github.com/Cosmian/tee-tools/blob/1717d380ce8065c46501a18ad22a982c5ca23362/sev_quote/src/quote.rs#L184

I will create a specific runner for that

Reply:
Done in c00096d152de059ca8b52dcb2b2d3c185ca9ed03.